packages/js/e2e-core-tests/src/specs/merchant/wp-admin-order-edit.test.js

```diff
@@ -69,14 +77,152 @@ const runEditOrderTest = () => {
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Add permission
+			await merchant.addDownloadableProductPermission( product2Name );
+
+			// Verify new downloadable product permission details
+			await merchant.verifyDownloadableProductPermission( product2Name )
+		} );
+
+		it( 'can edit downloadable product permissions', async () => {
+			// Define expected downloadable product attributes
+			const expectedDownloadsRemaining = '10';
+			const expectedDownloadsExpirationDate = '2050-01-01';
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Update permission
+			await merchant.updateDownloadableProductPermission(
+				productName,
+				expectedDownloadsExpirationDate,
+				expectedDownloadsRemaining
+			);
+
+			// Verify new downloadable product permission details
+			await merchant.verifyDownloadableProductPermission(
+				productName,
+				expectedDownloadsExpirationDate,
+				expectedDownloadsRemaining
+			);
+		} );
+
+		it( 'can revoke downloadable product permissions', async () => {
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Revoke permission
+			await merchant.revokeDownloadableProductPermission( productName );
+
+			// Verify
+			await expect( page ).not.toMatchElement( 'div.order_download_permissions', {
+				text: productName
+			} );
+		} );
+
+		it( 'should not allow downloading a product if download attempts are exceeded', async () => {
+			// Define expected download error reason
+			const expectedReason = 'Sorry, you have reached your download limit for this file';
+
+			// Setup data
+			productId = await createSimpleDownloadableProduct( productName, 0 );
+			orderId = await createOrder( {
+				productId,
+				customerBilling,
+				status: orderStatus.processing
+			} );
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Open download page
+			const downloadPage = await merchant.openDownloadLink();
+
+			// Verify file download cannot start
+	  		await merchant.verifyCannotDownloadFromBecause( downloadPage, expectedReason );
+		} );
+
+		it( 'should not allow downloading a product if expiration date is exceeded', async () => {
+			// Define expected download error reason
+			const expectedReason = 'Sorry, this download has expired';
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Update permission so that the expiration date has already passed
+			// Note: Seems this operation can't be performed through the API
+			await merchant.updateDownloadableProductPermission( productName, '2018-12-14' );
+
+			// Open download page
+			const downloadPage = await merchant.openDownloadLink();
+
+			// Verify file download cannot start
+	  		await merchant.verifyCannotDownloadFromBecause( downloadPage, expectedReason );
+		} );
+
+		afterEach( async () => {
+			await withRestApi.deleteAllOrders();
+			await withRestApi.deleteAllProducts();
+		} );
```

Comment:
@rodelgc If you're happy with this, I will also move this `afterEach` to the line 112 so that both `Describe` blocks follow the same structure.

Reply:
Regarding the location of the `afterEach` block, yep it would be nice to have it right after `beforeEach` although it isn't a strict requirement. Still, it's good to relocate it there for styling consistency.

Regarding deleting of all orders and products, I see why you went this approach. It's because some of the tests you added will create another order or product aside from the ones that `beforeEach` already created. I wonder if there's a more efficient way of cleaning them up without deleting all orders and products, and a way to set them up without creating duplicate orders/products in some of the new tests. I'm afraid that deleting all of them after each test would interfere with other tests when our test suite gained the ability to be run in parallel in the future.

Reply:
> Regarding the location of the afterEach block, yep it would be nice to have it right after beforeEach although it isn't a strict requirement. Still, it's good to relocate it there for styling consistency.

Indeed! I'll relocate it.

> Regarding deleting of all orders and products, I see why you went this approach. It's because some of the tests you added will create another order or product aside from the ones that beforeEach already created. I wonder if there's a more efficient way of cleaning them up without deleting all orders and products, and a way to set them up without creating duplicate orders/products in some of the new tests. I'm afraid that deleting all of them after each test would interfere with other tests when our test suite gained the ability to be run in parallel in the future.

Makes sense, if we expect the tests to be run in parallel, then we need to reduce potential risks that might compromise the test suite stability. I usually try to apply the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle, but parallel execution looks like a feature that will be surely added at some point, so I think it's good that we start preparing for it now in order to reduce the amount of technical debt later.

I'll check if we have any method like `withRestApi.deleteOrder( orderId )` or similar (and the same for products), that I could use for this purpose. If there isn't any, I will create them.

Thank you very much, @rodelgc!